pages/message-sent/index.tsx

```diff
@@ -16,6 +17,11 @@ export default function MessageSent() {
         sendAnalytics(BetterSocialEventTracking.SUCCESS_SCREEN_OPEN)
     }, [])
 
+    const onGoToProfileButtonClicked = () => {
+        const usernameToBeRedirected = localStorage.getItem(MessageEnum.targetUserName)
+        router.push(`/${usernameToBeRedirected}`)
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential issue with this code. It assumes that `localStorage.getItem(MessageEnum.targetUserName)` will always return a valid username. However, if the local storage does not contain a value for `MessageEnum.targetUserName`, `usernameToBeRedirected` will be `null` and the router will try to navigate to an invalid URL. This could lead to unexpected behavior or errors. 

Consider adding error handling to check if `usernameToBeRedirected` is not null before navigating.

```diff
  const onGoToProfileButtonClicked = () => {
-     const usernameToBeRedirected = localStorage.getItem(MessageEnum.targetUserName)
-     router.push(`/${usernameToBeRedirected}`)
+     const usernameToBeRedirected = localStorage.getItem(MessageEnum.targetUserName);
+     if (usernameToBeRedirected) {
+         router.push(`/${usernameToBeRedirected}`);
+     } else {
+         // Handle error or provide feedback to user
+     }
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->